### DISCUSSION TITLE
v2.0.2 Dockerfile only deployment running queue worker

### DISCUSSION THREAD

**Discussion by jericopulvera** [serversideup/docker-php#98] [Q&A] [answered]
I have this Dockerfile that is working on deployment and I want to add queue worker on it.

Based on the docs https://github.com/serversideup/docker-php#running-a-laravel-queue all you need to do is pass the Laravel Queue command to the container and S6 will automatically monitor it for you.

The example given on the docs is on docker-compose file. I have tried it on Dockerfile only using CMD and it runs the queue:work but the problem is the nginx stopped working.

I'm hoping someone knows the answer. I have also tried adding a queue:work command on a /etc/services.d/queue directory as suggested here https://community.fly.io/t/full-stack-laravel-bytes-tutorial-supervisor-not-working/6400/6 and it's also not working.

Dockerfile
```
# syntax = docker/dockerfile:experimental
ARG PHP_VERSION=8.0
FROM serversideup/php:${PHP_VERSION}-fpm-nginx-v2.0.2 as base

ARG PHP_VERSION

RUN apt-get update && apt-get install -y \
    git curl zip unzip rsync ca-certificates vim htop cron \
    php${PHP_VERSION}-mysql php${PHP_VERSION}-bcmath \
    php${PHP_VERSION}-swoole php${PHP_VERSION}-xml php${PHP_VERSION}-mbstring \
    php${PHP_VERSION}-intl php${PHP_VERSION}-memcache php${PHP_VERSION}-memcached \
    && apt-get clean \
    && rm -rf /var/lib/apt/lists/*

WORKDIR /var/www/html
# copy application code, skipping files based on .dockerignore
COPY . /var/www/html

RUN composer install --optimize-autoloader --no-dev \
    && mkdir -p storage/logs \
    && php artisan optimize:clear \
    && chown -R webuser:webgroup /var/www/html \
    && sed -i 's/protected \$proxies/protected \$proxies = "*"/g' app/Http/Middleware/TrustProxies.php \
    && echo "MAILTO=\"\"\n* * * * * webuser /usr/bin/php /var/www/html/artisan schedule:run" > /etc/cron.d/laravel \
    && rm -rf /etc/cont-init.d/* \
    && cp .fly/entrypoint.sh /entrypoint \
    && chmod +x /entrypoint

RUN rm /etc/nginx/sites-available/ssl-off \
    && cp .fly/nginx-default-2.0.2 /etc/nginx/sites-available/ssl-off

EXPOSE 8080

ENTRYPOINT ["/entrypoint"]

# CMD ["su", "webuser", "-c", "php artisan queue:work --tries=3"]
```

**Comment by jaydrogers** [comment, marked as answer]
I wonder if this has to do with you overriding the `ENTRYPOINT`?

If you notice in my base image, it requires `init`, which is S6 Overlay V3 standards: https://github.com/serversideup/s6-overlay/blob/main/Dockerfile#L48

I would remove your line (to inherit what is upstream) or set it explicitly match what I have.

**Reply by jericopulvera** [reply]
I have removed `ENTRYPOINT ["/entrypoint"]` and now it works without SSL_MODE=off.

Output without SSL_MODE in env

<img width="651" alt="image" src="https://user-images.githubusercontent.com/23246308/202836288-475d9bcb-e54c-4bec-8907-b98a52fce742.png">

Output with SSL_MODE=off

<img width="1011" alt="image" src="https://user-images.githubusercontent.com/23246308/202836324-240b82b7-6c0a-48be-8d46-e3fef0adae6b.png">

Here's the repo link if you're interested.
https://github.com/jericopulvera/dockerized-laravel-deployment

**Reply by jaydrogers** [reply]
Thank you for posting your code! I was able to replicate and diagnose the issue. 🙌

### The problem
You are setting a different port for NGINX to operate on:
https://github.com/jericopulvera/dockerized-laravel-deployment/blob/main/.fly/nginx-default-2.0.2#L2-L3

### Why this is bad
My healthcheck is expecting the server to listen on port 80: https://github.com/serversideup/docker-php/blob/main/src/fpm-nginx/etc/s6-overlay/s6-rc.d/nginx/data/check#L2

### Your options
1. Revert to port 80 (recommended if you can)
2. Overwrite my script to make sure it is health checking port 8080 after your modifications

**Reply by jericopulvera** [reply]
@jaydrogers Thanks that fixed it but what do you recommend for running additional commands on startup when when overriding entrypoint breaks things.

**Reply by jaydrogers** [reply]
I would NOT override any commands in the Dockerfile itself.

All commands should be passed by your orchestrator (Docker Compose, Docker Swarm, Kubernetes, etc). For example: https://docs.docker.com/compose/compose-file/#command
